File: php/features.php
<?php
declare(strict_types=1);

// GiveFood SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GiveFoodFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GiveFoodBaseFeature();
            case "test":
                return new GiveFoodTestFeature();
            default:
                return new GiveFoodBaseFeature();
        }
    }
}

File: py/features.py
# GiveFood SDK feature factory

from feature.base_feature import GiveFoodBaseFeature
from feature.test_feature import GiveFoodTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GiveFoodBaseFeature(),
        "test": lambda: GiveFoodTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
